src/SideNavigation/SideNavigation.js

```diff
@@ -1,204 +1,263 @@
+    selectedId: 'The id of the selected `SideNavListItem`.',
+    onItemSelect: 'Callback function when user selects an item .'
 };
 
-export const SideNavGroup = props => {
-    const { title, children, className, titleProps, ...rest } = props;
+export const SideNavListItem = ({children, expandedIds = [], glyph, id, isSubItem, name, onClick, onItemSelect, selected, selectedId, url, ...props}) => {
+    const getClasses = () => {
+        return classnames(
+            {
+                'fd-side-nav__link': !isSubItem,
+                'fd-side-nav__sublink': isSubItem,
+                'is-selected': selected,
+                'has-child': hasChild,
+                'is-expanded': expandedIds.includes(id)
+            }
+        );
+    };
 
-    const sideNavGroupClasses = classnames(
-        'fd-side-nav__group',
-        className
+    const renderLink = () => {
+        return (
+            <a
+                className={getClasses()}
+                href={url}
+                onClick={(e) => {
+                    onClick(e);
+                    onItemSelect(e, id, hasChild);
+                }}>
+                {glyph ? (
+                    <span
+                        className={`fd-side-nav__icon sap-icon--${glyph} sap-icon--l`}
+                        role='presentation' />
+                ) : null}
+                {name}
+            </a>
+        );
+    };
+
+    let hasChild = false;
+    React.Children.forEach(children, (child) => {
+        if (React.isValidElement(child) && child.type === SideNavSubList) {
+            hasChild = true;
+        }
+    });
+
+    return (
+        <li {...props}
+            className='fd-side-nav__item'
+            key={id}>
+            {url && renderLink()}
+            {React.Children.map(children, (child) => {
+                if (React.isValidElement(child) && child.type !== SideNavSubList) {
+                    return React.cloneElement(child, {
+                        children: (<React.Fragment>
+                            {glyph ? (
+                                <span
+                                    className={`fd-side-nav__icon sap-icon--${glyph} sap-icon--l`}
+                                    role='presentation' />
+                            ) : null}
+                            {child.props.children}
+                        </React.Fragment>),
+                        className: getClasses()
+                    });
+                } else if (React.isValidElement(child) && child.type === SideNavSubList) {
+                    return React.cloneElement(child, {
+                        onItemSelect: onItemSelect,
+                        open: expandedIds.includes(id),
+                        selectedId: selectedId
+                    });
+                } else {
+                    return child;
+                }
+            })}
+        </li>
     );
+};
 
+SideNavListItem.propTypes = {
+    children: PropTypes.node,
```

Comment:
Same comment as above regarding specific instances of children.

Reply:
After trying to tackle this, there is not a great way in React to enforce what components are passed as children. For now we are ignoring this issue per a conversation with @greg-a-smith.